rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -134,6 +134,7 @@ public AnnotatedType withType(@Nullable JavaType type) {
             return withTypeExpression(typeExpression.withType(type));
         }
 
+        @Deprecated
```

Comment:
Might be kind to add a note here to each of these deprecated classes why they where deprecated, and what to use now. Folks are likely to stumble across these deprecations in their existing recipes, and would otherwise be left wondering what to use instead.